classes/image_types_ostree.bbclass

```diff
@@ -104,26 +104,30 @@ IMAGE_CMD_ostree () {
         ln -sf var/roothome root
     fi
 
-    checksum=`sha256sum ${DEPLOY_DIR_IMAGE}/${OSTREE_KERNEL} | cut -f 1 -d " "`
-
-    cp ${DEPLOY_DIR_IMAGE}/${OSTREE_KERNEL} boot/vmlinuz-${checksum}
+    FILES_TO_CHECKSUM="${DEPLOY_DIR_IMAGE}/${OSTREE_KERNEL} ${INITRAMFS_IMAGE}-${MACHINE}.${INITRAMFS_FSTYPES} ${KAKUS}"
+    # calculate checksum of kernel and initramfs. In case of fitimage we ignore error becuase initramfs is not there
+    checksum=$(cat ${FILES_TO_CHECKSUM} 2>/dev/null  | sha256sum | cut -f 1 -d " ")
```

Comment:
Would it make sense to separate the checksum calculation out into three separate cases? If we already have two cases, and one of them is actually two cases in which an error is expected in one of them, it seems safer to just calculate it separately for all three:
1. fitImage case where there is just a kernel
2. "else" case where there is kernel and initramfs (no error ignoring necessary)
3. "else" case with devicetree

Reply:
I would agree that fewer variations are better, but since all three cases are different, I'd rather have three separate cases. In particular, I don't like that we ignore potential errors in the non-FIT case.

Reply:
ok I will fix it as soon as I am back to pc